bundle/Core/SignalSlot/AbstractPublishSlot.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Netgen\TagsBundle\Core\SignalSlot;
+
+use eZ\Publish\API\Repository\Exceptions\NotFoundException;
+use eZ\Publish\Core\SignalSlot\Signal;
+use EzSystems\PlatformHttpCacheBundle\PurgeClient\PurgeClientInterface;
+use EzSystems\PlatformHttpCacheBundle\SignalSlot\AbstractSlot;
+use Netgen\TagsBundle\Core\Persistence\Legacy\Tags\Handler;
+use Netgen\TagsBundle\SPI\Persistence\Tags\Tag;
+
+abstract class AbstractPublishSlot extends AbstractSlot
+{
+    /**
+     * @var Handler
+     */
+    protected $tagsHandler;
+
+    /**
+     * AbstractPublishSlot constructor.
+     *
+     * @param PurgeClientInterface $purgeClient
+     * @param Handler $tagsHandler
+     */
+    public function __construct(PurgeClientInterface $purgeClient, Handler $tagsHandler)
+    {
+        parent::__construct($purgeClient);
+        $this->tagsHandler = $tagsHandler;
+    }
+
+    /**
+     * @param Signal $signal
+     *
+     * @throws NotFoundException
+     *
+     * @return array
+     */
+    protected function generateTags(Signal $signal)
+    {
+        $ngTagId = $this->getNgTagId($signal);
+
+        $ngTag = $this->tagsHandler->load($ngTagId);
+
+        return $this->getNgTagTags($ngTag);
+    }
+
+    /**
+     * Extracts tag id from signal.
+     *
+     * @param Signal $signal
+     *
+     * @return mixed
+     */
+    abstract protected function getNgTagId(Signal $signal);
```

Comment:
There's no need to prefix methods with `Ng`. The namespace already contains `Netgen`, so no need for it in the method names :)